src/main/java/org/openstreetmap/atlas/utilities/graphs/DirectedAcyclicGraph.java

```diff
@@ -136,6 +138,50 @@ public boolean isSource(final V vertex)
         return this.inMap.get(vertex).isEmpty();
     }
 
+    /**
+     * @return The ordered groups of vertices that all belong to the same priority level within the
+     *         DAG
+     */
+    public List<Set<V>> processGroups()
+    {
+        final Stack<Set<V>> stack = new Stack<>();
+        stack.push(new HashSet<>(getSinks()));
+        final Set<V> added = new HashSet<>(getSinks());
+        final Set<V> sourcesNotAdded = new HashSet<>(getSources());
+        while (!sourcesNotAdded.isEmpty())
+        {
+            final Set<V> allCandidates = new HashSet<>();
+            final Set<V> candidates = new HashSet<>();
```

Comment:
This is a minor nit: but it may be helpful for future readers if these two variables had more distinguishing names. `allCandidates` and `candidates` may not be immediately clear.